File: Android/SoundRecorder/src/com/wiseapps/davacon/core/se/SEAudioStream.java
//package com.wiseapps.davacon.core.se;
//
//import android.content.Context;
//
///**
// * @author varya.bzhezinskaya@wise-apps.com
// *         Date: 4/14/14
// *         Time: 11:53 AM
// *
// * Set of public methods could not be changed!!!
// */
//public abstract class SEAudioStream {
//
//    final Context context;
//
//    public Mode mode;
//
//    public static enum Mode {
//        READ,
//        WRITE
//    }
//
//    protected SEAudioStream(Context context) {
//        this.context = context;
//    }
//
//    /**
//     * Opens the stream
//     */
//    abstract void open(Mode mode);
//
//    /**
//     * Closes the stream
//     */
//    abstract void close();
//
//    /**
//     * Clears the stream
//     */
//    abstract void clear();
//
//    /**
//     * Writes the data to the end of the stream
//     *
//     * @param data data to be appended
//     */
//    abstract void write(byte[] data);
//
//    /**
//     * Reads data from the stream
//     *
//     * @param position position to start reading from
//     * @param duration duration of the data to be read
//     * @return read data
//     */
//    abstract byte[] read(double position, double duration);
//
//    abstract Mode getMode();
//}

File: Blackberry/SoundRecorder/src/core/se/SEAudioStream.cpp
/*
 * SEAudioStream.cpp
 *
 *  Created on: 16.04.2014
 *      Author: Timofey Kovalenko <timothy.kovalenko@wise-apps.com>
 */

#include "SEAudioStream.h"
#include "utils/Loger.h"

SEAudioStream::SEAudioStream() : mode(modeUnknown),
                                 format(formatUnknown),
                                 currentPositionMills(0),
                                 startPositionInMiils(0),
                                 durationInMills(0){}

SEAudioStream::~SEAudioStream() {}

//int SEAudioStream::getMode() {
//	Loger::Debug(typeid(this).name(), "getMode");
//	return mode;
//}

bool SEAudioStream::open(SEAudioStreamMode mode) {
	Loger::Debug(typeid(this).name(), "open");
	return false;
}

void SEAudioStream::close() {
	Loger::Debug(typeid(this).name(), "close");
}

bool SEAudioStream::clear() {
	Loger::Debug(typeid(this).name(), "clear");
	return false;
}

bool SEAudioStream::write(char *data) {
	Loger::Debug(typeid(this).name(), "write");
	return false;
}

unsigned int SEAudioStream::read(char *data, unsigned int position, unsigned int duration) {
	Loger::Debug(typeid(this).name(), "read");
	return 0;
}
